File: threads/src/main/java/im/threads/internal/holders/ConsultIsTypingViewHolderNew.java
package im.threads.internal.holders;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;
import im.threads.ChatStyle;
import im.threads.R;
import im.threads.internal.Config;

/**
 * layout/item_consult_typing.xml
 */
public final class ConsultIsTypingViewHolderNew extends RecyclerView.ViewHolder {
    public ImageView mConsultAvatar;
    private TextView mViewTypingInProgress;
    private ChatStyle style;

    public ConsultIsTypingViewHolderNew(ViewGroup parent) {
        super((LayoutInflater.from(parent.getContext())).inflate(R.layout.item_consult_typing, parent, false));
        mConsultAvatar = itemView.findViewById(R.id.image);
        mViewTypingInProgress = itemView.findViewById(R.id.typing_in_progress);
        if (style == null) style = Config.instance.getChatStyle();
        mViewTypingInProgress.setTextColor(ContextCompat.getColor(itemView.getContext(), style.chatSystemMessageTextColor));
        mConsultAvatar.getLayoutParams().height = (int) itemView.getContext().getResources().getDimension(style.operatorSystemAvatarSize);
        mConsultAvatar.getLayoutParams().width = (int) itemView.getContext().getResources().getDimension(style.operatorSystemAvatarSize);
    }

    public void onBind(View.OnClickListener consultClickListener) {
        mConsultAvatar.setOnClickListener(consultClickListener);
    }
}

File: threads/src/main/java/im/threads/internal/holders/ConsultIsTypingViewHolderNew.kt
package im.threads.internal.holders

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import im.threads.ChatStyle
import im.threads.R
import im.threads.internal.Config
import im.threads.internal.image_loading.ImageModifications
import im.threads.internal.image_loading.loadImage
import im.threads.internal.model.ConsultTyping
import im.threads.internal.utils.FileUtils

/**
 * layout/item_consult_typing.xml
 */
class ConsultIsTypingViewHolderNew(parent: ViewGroup) : RecyclerView.ViewHolder(
    LayoutInflater.from(parent.context).inflate(R.layout.item_consult_typing, parent, false)
) {
    private val style: ChatStyle = Config.instance.chatStyle

    private val mConsultAvatar = itemView.findViewById<ImageView>(R.id.image).apply {
        layoutParams.height =
            itemView.context.resources.getDimension(style.operatorSystemAvatarSize)
                .toInt()
        layoutParams.width =
            itemView.context.resources.getDimension(style.operatorSystemAvatarSize)
                .toInt()
    }

    init {
        itemView.findViewById<TextView>(R.id.typing_in_progress).apply {
            setTextColor(
                ContextCompat.getColor(
                    itemView.context,
                    style.chatSystemMessageTextColor
                )
            )
        }
    }

    fun onBind(consultTyping: ConsultTyping, consultClickListener: View.OnClickListener) {
        mConsultAvatar.setOnClickListener(consultClickListener)
        mConsultAvatar.setImageResource(style.defaultOperatorAvatar)
        mConsultAvatar.loadImage(
            FileUtils.convertRelativeUrlToAbsolute(consultTyping.avatarPath),
            listOf(ImageView.ScaleType.FIT_XY, ImageView.ScaleType.CENTER_CROP),
            modifications = listOf(ImageModifications.CircleCropModification)
        )
    }
}
